circuits-circom/venmo_receive.circom

```diff
@@ -0,0 +1,146 @@
+    EV.precomputed_sha <== precomputed_sha;
+    EV.in_body_padded <== in_body_padded;
+    EV.in_body_len_padded_bytes <== in_body_len_padded_bytes;
+    signal header_hash[256] <== EV.sha;
+
+    modulus_hash <== EV.pubkey_hash;
+
+    // FROM HEADER REGEX
+    // This extracts the from email, and the precise regex format can be viewed in the README
+    var max_email_from_packed_bytes = count_packed(max_email_from_len, pack_size);
+    assert(max_email_from_packed_bytes < max_header_bytes);
+
+    signal input email_from_idx;
+    signal output reveal_email_from_packed[max_email_from_packed_bytes]; // packed into 7-bytes
+
+    signal (from_regex_out, from_regex_reveal[max_header_bytes]) <== FromRegex(max_header_bytes)(in_padded);
+    from_regex_out === 1;
+    reveal_email_from_packed <== ShiftAndPackMaskedStr(max_header_bytes, max_email_from_len, pack_size)(from_regex_reveal, email_from_idx);
+
+    // VENMO RECEIVE AMOUNT REGEX
+    var max_email_amount_packed_bytes = count_packed(max_email_amount_len, pack_size);
+    assert(max_email_amount_packed_bytes < max_header_bytes);
+
+    signal input venmo_amount_idx;
+    signal output reveal_email_amount_packed[max_email_amount_packed_bytes]; // packed into 7-bytes. TODO: make this rotate to take up even less space
+
+    signal amount_regex_out, amount_regex_reveal[max_header_bytes];
+    (amount_regex_out, amount_regex_reveal) <== VenmoReceiveAmountRegex(max_header_bytes)(in_padded);
+    amount_regex_out === 1;
+
+    reveal_email_amount_packed <== ShiftAndPackMaskedStr(max_header_bytes, max_email_amount_len, pack_size)(amount_regex_reveal, venmo_amount_idx);
+
+    // TIMESTAMP REGEX
+    var max_email_timestamp_packed_bytes = count_packed(max_email_timestamp_len, pack_size);
+    assert(max_email_timestamp_packed_bytes < max_header_bytes);
+
+    signal input email_timestamp_idx;
+    signal output reveal_email_timestamp_packed[max_email_timestamp_packed_bytes]; // packed into 7-bytes
+
+    signal timestamp_regex_out, timestamp_regex_reveal[max_header_bytes];
+    (timestamp_regex_out, timestamp_regex_reveal) <== VenmoTimestampRegex(max_header_bytes)(in_padded);
+    timestamp_regex_out === 1;
+
+    // PACKING
+    reveal_email_timestamp_packed <== ShiftAndPackMaskedStr(max_header_bytes, max_email_timestamp_len, pack_size)(timestamp_regex_reveal, email_timestamp_idx);
+    
+    // VENMO RECEIVE PAYEE ID AND PAYER ID REGEX
+    var max_venmo_id_packed_bytes = count_packed(max_payee_len, pack_size); // ceil(max_num_bytes / 7)
+    
+    signal input venmo_payee_id_idx;
+    signal reveal_payee_packed[max_venmo_id_packed_bytes];
+    signal input venmo_payer_id_idx;
+    signal reveal_payer_packed[max_venmo_id_packed_bytes];
+
+    signal (receive_regex_out, payer_regex_reveal[max_body_bytes], payee_regex_reveal[max_body_bytes]) <== VenmoReceiveIdRegex(max_body_bytes)(in_body_padded);
+    signal is_found_receive <== IsZero()(receive_regex_out);
+    is_found_receive === 0;
```

Comment:
Should this be instead 2 (or >2). To ensure we found at least 2 matches for the 2 ids.